modules/hal_rpi_pico/pico/config_autogen.h

```diff
@@ -22,7 +22,11 @@
  * BUILD_ASSERT is included through toolchain.h.
  */
 #include <zephyr/toolchain.h>
+#ifndef __cplusplus
+#ifndef static_assert
```

Comment:
nested ifdefs should be avoided when possible.

```suggestion
#if !defined(__cplusplus) && !defined(static_assert)
```